src/move/econia/sources/market.move

```diff
@@ -3057,16 +3256,18 @@ module econia::market {
                     quote_withdraw, underwriter_id);
             // Declare return assignment variable.
             let self_match_cancel;
+            // Declare empty fill event queue inside of an option.
+            let fill_event_queue_option = option::some(vector[]);
             // Match against order book, storing optionally modified
             // asset inputs, base and quote trade amounts, quote fees
             // paid, and if a self match requires canceling the rest of
-            // the order.
+            // the order. (Increments order book counter).
```

Comment:
why?

Reply:
Previously, the order book counter was only incremented when a limit order resulted in a post to the book

Per this PR, the order book counter is also incremented whenever the matching engine is called so that fill events can be assigned a unique order ID